projects/ngx-sub-form/src/lib/ngx-sub-form-utils.ts

```diff
@@ -32,6 +35,30 @@ export type FormErrors<FormInterface> = null | Partial<
   }
 >;
 
+// using set/patch value options signature from form controls to allow typing without additional casting
+export interface TypedAbstractControl<TValue> extends AbstractControl {
+  value: TValue;
+  valueChanges: Observable<TValue>;
+  setValue(value: TValue, options?: Parameters<FormControl['setValue']>[1]): void;
```

Comment:
```suggestion
  setValue(value: TValue, options?: Parameters<AbstractControl['setValue']>[1]): void;
```

Reply:
This was actually done on purpose, I will revert to using  `AbstractControl.setValue` here, but just to highlight the reasoning and love to hear your input [see here](https://github.com/cloudnc/ngx-sub-form/pull/139#issuecomment-590354658).